core/banking/checkpoint.go

```diff
@@ -135,15 +135,19 @@ func (e *Engine) getRecurringTransfers() *checkpoint.RecurringTransfers {
 }
 
 func (e *Engine) getScheduledTransfers() []*checkpoint.ScheduledTransferAtTime {
-	out := []*checkpoint.ScheduledTransferAtTime{}
+	out := make([]*checkpoint.ScheduledTransferAtTime, 0, len(e.scheduledTransfers))
 
 	for k, v := range e.scheduledTransfers {
 		transfers := make([]*checkpoint.ScheduledTransfer, 0, len(v))
 		for _, v := range v {
 			transfers = append(transfers, v.ToProto())
 		}
 
-		out = append(out, &checkpoint.ScheduledTransferAtTime{DeliverOn: k.Unix(), Transfers: transfers})
+		// k is a Unix nano timestamp, and we want a Unix timestamp.
+		// So we divide it by (time.Second / time.Nanosecond).
+		deliverOnAsUnix := k / 1e9
+
+		out = append(out, &checkpoint.ScheduledTransferAtTime{DeliverOn: deliverOnAsUnix, Transfers: transfers})
```

Comment:
does it matter here that it's a unix seconds or nanoseconds? I'd rather keep it all in nano to be fair as it shall fit in the int64. Also wouldn't restoring the checkpoint here be broken then?

Reply:
hum right... because ultimately we will need to reuse the checkpoint form mainnet which have transfers
alright, makes sense. however instead of using 1e9 here why not just:
```
k / time.Nanosecond
```
or 
```
deliverOnAsUnix := time.Unix(0, k).Unix()
```

Reply:
In golang, the `time.<Nano|Micro|Milli>second` works the other way around. The smallest measure `time.Nanosecond` is equal to `1`, and it multiplies by `1000` for each upper tranche, so:

- `k / time.Nanosecond`  is basically `k / 1` that won't do it.
- `time.Unix(0, k).Unix()` would work but just more "expensive". A division should always work for us.

Reply:
right, typo, I mean time.Second, sorry to be picky, but that just makes it too cryptic, like anywhere else when you want a time in second your would do 4*time.Second or in Nano 41*time.Nanosecond, here just use it too.

Reply:
OK